src/constraint/encoding.ts

```diff
@@ -375,6 +375,18 @@ export const FIELD_CONSTRAINTS: EncodingConstraintModel<FieldQuery>[] = [
       }
       return true;
     }
+  },{
+    name: 'onlyCountWithAutoCountFieldOnSort',
+    description: 'Sort by * if and only if op is count',
+    properties: [Property.SORT, getEncodingNestedProp('sort', 'field'), getEncodingNestedProp('sort', 'op')],
+    allowWildcardForProperties: false,
+    strict: true,
+    satisfy: (fieldQ: FieldQuery, _: Schema, __: PropIndex<Wildcard<any>>, ___: QueryConfig) => {
+      if (fieldQ.sort && !!(fieldQ.sort as SortField).field) {
```

Comment:
Don't hack using `as SortField` -- do proper type guard using `isSortField` in VL's sort.ts